File: skills/gemini-batch/scripts/check_status.py
#!/usr/bin/env python3
"""Check batch job status.

Usage:
    python check_status.py <job_name>
    python check_status.py batches/abc123 --wait

Requirements:
    pip install google-genai
"""

import argparse
import os
import sys
import time

# Load .env file if present
try:
    from dotenv import load_dotenv

    load_dotenv()
except ImportError:
    pass


def get_client():
    """Get Gemini API client."""
    try:
        from google import genai
    except ImportError:
        print("Error: google-genai not installed. Run: pip install google-genai")
        sys.exit(1)

    api_key = os.environ.get("GOOGLE_API_KEY") or os.environ.get("GEMINI_API_KEY")
    if not api_key:
        print("Error: Set GOOGLE_API_KEY or GEMINI_API_KEY environment variable")
        sys.exit(1)

    return genai.Client(api_key=api_key)


def check_status(job_name: str, wait: bool = False) -> str:
    """Check batch job status.

    Args:
        job_name: The batch job name/ID
        wait: Whether to poll until completion

    Returns:
        Final job state
    """
    client = get_client()

    completed_states = {
        "JOB_STATE_SUCCEEDED",
        "JOB_STATE_FAILED",
        "JOB_STATE_CANCELLED",
        "JOB_STATE_EXPIRED",
    }

    batch_job = client.batches.get(name=job_name)
    state = (
        batch_job.state.name
        if hasattr(batch_job.state, "name")
        else str(batch_job.state)
    )

    if wait:
        print(f"Polling status for job: {job_name}")
        while state not in completed_states:
            print(f"Current state: {state}")
            time.sleep(30)
            batch_job = client.batches.get(name=job_name)
            state = (
                batch_job.state.name
                if hasattr(batch_job.state, "name")
                else str(batch_job.state)
            )

    if state == "JOB_STATE_SUCCEEDED":
        print("Job succeeded!")
    elif state == "JOB_STATE_FAILED":
        print("Job failed!")
        if hasattr(batch_job, "error") and batch_job.error:
            print(f"Error: {batch_job.error}")
    else:
        print(f"Job state: {state}")

    return state


def main():
    parser = argparse.ArgumentParser(description="Check batch job status")
    parser.add_argument("job_name", help="Batch job name/ID")
    parser.add_argument(
        "--wait",
        "-w",
        action="store_true",
        help="Wait for job to complete (polls every 30s)",
    )

    args = parser.parse_args()

    try:
        check_status(job_name=args.job_name, wait=args.wait)
    except Exception as e:
        print(f"Error: {e}")
        sys.exit(1)


if __name__ == "__main__":
    main()

File: skills/gemini-batch/scripts/check_status.js
#!/usr/bin/env node
"use strict";

import process from "node:process";
import { parseArgs } from "node:util";

import dotenv from "dotenv";
import { GoogleGenAI } from "@google/genai";

dotenv.config();

const COMPLETED_STATES = new Set([
  "JOB_STATE_SUCCEEDED",
  "JOB_STATE_FAILED",
  "JOB_STATE_CANCELLED",
  "JOB_STATE_EXPIRED",
]);

function printUsage() {
  console.log(`Check batch job status.

Usage:
  node scripts/check_status.js <job_name>
  node scripts/check_status.js batches/abc123 --wait

Options:
  --wait, -w      Poll every 30s until complete
  --help, -h      Show this help message

Requirements:
  npm install @google/genai@latest dotenv@latest
`);
}

function getApiKey() {
  const apiKey = process.env.GOOGLE_API_KEY || process.env.GEMINI_API_KEY;
  if (!apiKey) {
    throw new Error("Set GOOGLE_API_KEY or GEMINI_API_KEY environment variable");
  }
  return apiKey;
}

function getState(value) {
  if (!value) return "UNKNOWN";
  if (typeof value === "string") return value;
  if (typeof value.name === "string") return value.name;
  return String(value);
}

function sleep(ms) {
  return new Promise((resolve) => setTimeout(resolve, ms));
}

async function checkStatus({ jobName, wait }) {
  const ai = new GoogleGenAI({ apiKey: getApiKey() });

  let batchJob = await ai.batches.get({ name: jobName });
  let state = getState(batchJob.state);

  if (wait) {
    console.log(`Polling status for job: ${jobName}`);
    while (!COMPLETED_STATES.has(state)) {
      console.log(`Current state: ${state}`);
      await sleep(30000);
      batchJob = await ai.batches.get({ name: jobName });
      state = getState(batchJob.state);
    }
  }

  if (state === "JOB_STATE_SUCCEEDED") {
    console.log("Job succeeded!");
  } else if (state === "JOB_STATE_FAILED") {
    console.log("Job failed!");
    if (batchJob.error) {
      console.log(`Error: ${typeof batchJob.error === "string" ? batchJob.error : JSON.stringify(batchJob.error)}`);
    }
  } else {
    console.log(`Job state: ${state}`);
  }

  return state;
}

function parseCli() {
  const { values, positionals } = parseArgs({
    options: {
      wait: { type: "boolean", short: "w", default: false },
      help: { type: "boolean", short: "h", default: false },
    },
    allowPositionals: true,
  });

  if (values.help) {
    printUsage();
    process.exit(0);
  }

  if (positionals.length === 0) {
    printUsage();
    process.exit(1);
  }

  return {
    jobName: positionals[0],
    wait: values.wait,
  };
}

async function main() {
  try {
    const options = parseCli();
    await checkStatus(options);
  } catch (error) {
    console.error(`Error: ${error instanceof Error ? error.message : String(error)}`);
    process.exit(1);
  }
}

await main();
